pkg/kubelet/status/manager.go

```diff
@@ -247,6 +247,9 @@ func (m *manager) updateStatusInternal(pod *api.Pod, status api.PodStatus, force
 		oldStatus = pod.Status
 	}
 
+	// update observed generation
+	status.ObservedGeneration = pod.ObjectMeta.Generation
```

Comment:
I'd like to report observed generation for every sync of a PodSpec. This seems like a good place to do it.